packaging/resource_suite_s3_cache.py

```diff
@@ -608,3 +635,237 @@ def test_multiple_s3_endpoints_replication_issue1858(self):
+            admin_session.assert_icommand("iadmin mkresc cacheResc 'unixfilesystem' " + hostname + ":" + irods_config.irods_directory + "/cacheRescVault", 'STDOUT_SINGLELINE', 'cacheResc')
+            admin_session.assert_icommand('iadmin mkresc archiveResc s3 '+hostname+':/'+self.s3bucketname+'/irods/Vault "'+s3params+'"', 'STDOUT_SINGLELINE', 'archiveResc')
+            admin_session.assert_icommand("iadmin addchildtoresc demoResc cacheResc cache")
+            admin_session.assert_icommand("iadmin addchildtoresc demoResc archiveResc archive")
+
+    def tearDown(self):
+        super(Test_S3_Cache_Glacier_Base, self).tearDown()
+
+        # delete s3 bucket
+        try:
+            httpClient = urllib3.poolmanager.ProxyManager(
+                os.environ['http_proxy'],
+                timeout=urllib3.Timeout.DEFAULT_TIMEOUT,
+                cert_reqs='CERT_REQUIRED',
+                retries=urllib3.Retry(
+                    total=5,
+                    backoff_factor=0.2,
+                    status_forcelist=[500, 502, 503, 504]
+                )
+            )
+        except KeyError:
+            httpClient = None
+
+        if self.proto == 'HTTPS':
+            s3_client = Minio(self.s3endPoint,
+                              access_key=self.aws_access_key_id,
+                              secret_key=self.aws_secret_access_key,
+                              http_client=httpClient,
+                              region=self.s3region)
+        else:
+            s3_client = Minio(self.s3endPoint,
+                              access_key=self.aws_access_key_id,
+                              secret_key=self.aws_secret_access_key,
+                              http_client=httpClient,
+                              region=self.s3region,
+                              secure=False)
+
+        objects = s3_client.list_objects(self.s3bucketname, recursive=True)
+
+        if not hasattr(self, 'static_bucket_name'):
+            s3_client.remove_bucket(self.s3bucketname)
+
+        # tear down resources
+        with session.make_session_for_existing_admin() as admin_session:
+            admin_session.assert_icommand("iadmin rmchildfromresc demoResc archiveResc")
+            admin_session.assert_icommand("iadmin rmchildfromresc demoResc cacheResc")
+            admin_session.assert_icommand("iadmin rmresc archiveResc")
+            admin_session.assert_icommand("iadmin rmresc cacheResc")
+            admin_session.assert_icommand("iadmin rmresc demoResc")
+            admin_session.assert_icommand("iadmin modresc origResc name demoResc", 'STDOUT_SINGLELINE', 'rename', input='yes\n')
+
+        shutil.rmtree(IrodsConfig().irods_directory + "/cacheRescVault", ignore_errors=True)
+
+    def get_resource_context(self, resc_name):
+
+        resource_context = self.admin.run_icommand('iquest "%s" "SELECT RESC_CONTEXT where RESC_NAME = \'{resc_name}\'"'.format(**locals()))[0].strip()
+        return resource_context
```

Comment:
The body could be a one liner - `return self.admin.run_icommand( ...`

Reply:
Done for all